### ISSUE TITLE
3.0.0.1_B18 - All Up, No Down

### ISSUE THREAD

**Issue by MoonPie75** [BiglySoftware/BiglyBT#2536]
An interesting issue has suddenly arisen after the update to 3.0.0.1_B18. This is now happening:

<img width="302" alt="Screenshot" src="https://user-images.githubusercontent.com/102927788/164768904-0ab00a6d-109b-4cd2-ba8c-ac27dae6e7af.png">

Torrents start fine, download at expected speed for an hour or two, and then everything just stops downloading. All seeds and peers remain connected and uploading proceeds as normal. I just suddenly don't get any download. Stopping and restarting the torrents results in downloads coming back but the process just repeats itself after a while.

I'm not sure if this is related to B18 or not but I never saw it until the update. I haven't made any changes to BiglyBT or my system configuration going back at least a week before the problem appeared.

Any clues?

Thanks!

Java 11.0.14.1 (64 bit)
  Ubuntu
/usr/lib/jvm/java-11-openjdk-amd64

SWT v4942r22, gtk/3.24.20, zoom=100, dpi=120
Linux v5.13.0-40-generic, amd64 (64 bit)
B3.0.0.1_B18/4 az3

**Comment by parg** [comment]
Check the peers you are connected to in the "Peers" sub-tab. What is their completion?

**Comment by parg** [comment]
Also check if they are choking you - i.e. not accepting requests for download blocks. You can enable the "out reqs" column to see how many outstanding requests you have to a peer.

There's nothing new in B18 that should affect downloading.

**Comment by parg** [comment]
Do you have a global download limit set? If so check that

Options->Transfer: Use request throttling instead of delayed reads to limit download speed [no effect when download speed is unlimited]

is enabled.

Back later!

**Comment by MoonPie75** [comment]
Thanks parg.

The completion percentage varies among the peers but there are plenty that above mine. Also, the seeds stop downloading as well so ??? It's also interesting that this happens to all torrents at the same time. I've never actually been watching it as it happened, but I have once checked on it to see everything was fine and then looked back a couple of minutes later to find everything with no download.

I don't have out reqs enabled but I did look at snubbing, choking, and interest and there's nothing unusual there. I'll enable out reqs and report back next time it happens.

I do have a global download limit set at 73,242 on a commercial-grade, dedicated 1G fiber link but it's been like that for a couple of years without issue. 

I do have request throttling enabled. I'll turn that off and note any change.

I guess I did lie a little bit about system changes. I forgot that I have auto updates turned on in Ubuntu so there may have been some updates to the system that I didn't pay attention to. Maybe I'll try to roll back to a snapshot from a month or so ago and see what happens.

My downloads for week are now done so I won't be bringing anything more down until sometime next week. I'll update once I have more data.

Thanks!

**Comment by parg** [comment]
Strange stuff! 

I'd leave "request throttling" enabled, I've seen some odd behaviour on occasion with it disabled, was just checking that you hadn't disabled it for some reason.

**Comment by MoonPie75** [comment]
Hi parg,

I owe you an apology. We found the problem and it had nothing to do with BiglyBT.

Turns out that our local network genius recently installed a new anti-Denial-of-Service package onto our router without either testing it or bothering to tell anybody about it. The router has been classing inbound torrent traffic as a DoS flood and redirecting it to a null address based on some funky IP/port-based rules. That left everything but BiglyBT running fine so I never suspected that it was a network problem rather than something in the B18 update.

The short-term fix was to roll back the router to something that actually works. The long-term fix was to drive the network guy to the local zoo, smear him with bacon fat, and chuck him into the lion cage. I believe this combination will produce a permanent solution to the problem.

I'm very sorry to have wasted your time and mental energy on this. I should know better than to assume that local stuff is working correctly before posting. I'll be better with the due diligence before bothering you again.

Thanks for everything. It's much appreciated.
